components/results_table/commons/results_table_base.lua

```diff
@@ -271,20 +288,26 @@ end
 function BaseResultsTable:buildTeamOpponentConditions()
 	local config = self.config
 
-	local rawOpponentTemplate = Team.queryRaw(config.opponent) or {}
-	local opponentTemplate = rawOpponentTemplate.historicaltemplate or rawOpponentTemplate.templatename
-	if not opponentTemplate then
-		error('Missing team template for team: ' .. config.opponent)
-	end
+	local opponents = Array.append(config.aliases, config.opponent)
 
-	local opponentTeamTeplates = Team.queryHistorical(opponentTemplate) or {opponentTemplate}
+	local opponentTeamTemplates = {}
+
+	for _, opponent in pairs(opponents) do
+		local rawOpponentTemplate = Team.queryRaw(opponent) or {}
+		local opponentTemplate = rawOpponentTemplate.historicaltemplate or rawOpponentTemplate.templatename
+		if not opponentTemplate then
+			error('Missing team template for team: ' .. opponent)
+		end
```

Comment:
Why is this needed both here and L95?

Reply:
L95 gathers all TTs that are based on the main opponent (e.g. pagename). so non-aliased.
The list is later used to check whether a given TT is from an alias or not, which will enable the teamname in addition to the logo.

This line gathers all TTs that need to be queried for, including passed aliases that aren't redirects.

Reply:
Extract it to a function

Reply:
Extract the common part between L296 and L95 ;)

Reply:
done